pkg/kfilefields/bpf/filefields.bpf.c

```diff
@@ -64,6 +65,7 @@ int BPF_KRETPROBE(ig_fget_x, struct file *ret)
 	if (!ff)
 		return 0;
 
+	ff->f_inode = (u64)BPF_CORE_READ(ret, f_inode);
```

Comment:
Does it work, do you get the `struct inode*` from the underlying filesystem? Is it the same inode pointer when checking two different containers using the same image?

Or do you have to access with this?
```
ff->private_data = (u64)BPF_CORE_READ(ret, private_data);
struct ovl_inode *inode = (struct ovl_inode *) ff->private_data;
ff->f_inode = (u64)BPF_CORE_READ(inode, vfs_inode);
```

Note that overlay is a kernel module, so it depends on https://github.com/cilium/ebpf/pull/1300 but we recently updated to cilium/ebpf v0.14.0 with it. We'd still need to use `ProgramOptions` with `KernelModuleTypes` I think.

Reply:
With this method, we may need to provide the BTF for all potentially used file systems, or at least for the necessary structures.

Reply:
For `--host`, there may have many diffrent FS, other than btrfs/overlayfs.